cpp/oneapi/dal/io/csv/common.hpp

```diff
@@ -65,13 +68,19 @@ class ONEDAL_EXPORT data_source_base : public base {
 using v1::data_source_tag;
 using v1::data_source_impl;
 using v1::data_source_base;
+using v1::is_valid_float_v;
 
 } // namespace detail
 
 namespace v1 {
 
+template <typename Float = float>
 class data_source : public detail::data_source_base {
+    static_assert(detail::is_valid_float_v<Float>);
```

Comment:
I assume it would be better to move this line inside the constructors.

Reply:
Have put it here following the convention in algo's descriptor definitions. Besides, since static_asserts are checked during compile time, why is it better to move it inside the constructors?

Reply:
My assumption was that this code might be optimized out by the compiler if it's not on the main execution path.
Now I see that the code is in alignment with other algo's descriptors that are covered by various tests.
So it is better to leave this code as is.